@types/pal/image.d.ts

```diff
@@ -0,0 +1,64 @@
+declare module 'pal/image' {
+    export type ImageSource = import('pal/image/types').ImageSource;
+    export type IMemoryImageSource = import('pal/image/types').IMemoryImageSource;
+    export type PixelFormat = import('cocos/asset/assets/asset-enum').PixelFormat;
+    export class ImageData {
```

Comment:
Why not call it `Image` directly? ImageData seems just raw data like https://developer.mozilla.org/en-US/docs/Web/API/ImageData .
`Image` includes `data` property is fine. But `ImageData` also contains `data` property which is confused in my opinion.

Reply:
![image](https://github.com/cocos/cocos-engine/assets/5205637/2601cd15-c847-4dd4-8b28-65b9862e30a5)
They are defined in Dom.

There are currently two types of data, one is imagesource (HtmlImageElement) and the other is Raw image data, both of which currently need to exist.

The data attribute prefers to be compatible, and I understand that it should be named source would be better.

Reply:
DOM's Image is in global namespace, but our encapsulated `Image` could be in engine's namespace, for example `cc`. Will the class names in different namespace cause `conflict`?